src/server/network/network.ts

```diff
@@ -85,56 +93,86 @@ export class UDPServer {
     return () => this.clients.delete(client);
   }
 
-  // Set up callbacks for a given team and then bind the team to its multicast address and port
-  private initialiseListener = (listener: Listener) => {
-    listener.server.on('listening', () => {
+  // One of the UDP servers received a new message, send it on to all clients
+  private onMessage = (
+    team: TeamData,
+    packet: Buffer,
+    rinfo: dgram.RemoteInfo,
+  ) => {
+    for (let emit_cb of this.clients.values()) {
+      emit_cb('udp_packet', {
+        payload: packet,
+        team: {
+          name: team.name,
+          address: team.address,
+          port: team.port,
+        },
+        rinfo: {
+          family: rinfo.family,
+          address: rinfo.address,
+          port: rinfo.port,
+        },
+      });
+    }
+  };
+}
+
+class UDPServer {
+  private socket: dgram.Socket;
+
+  constructor(
+    private team: TeamData,
+    private msg_cb: MessageCallback,
+    ipv6: boolean,
+  ) {
+    if (ipv6) {
+      this.socket = dgram.createSocket({ type: 'udp6', reuseAddr: true });
+    } else {
+      this.socket = dgram.createSocket({ type: 'udp4', reuseAddr: true });
+    }
+
+    this.socket.on('listening', () => {
       // Add multicast membership if we are using a multicast address
-      if (listener.ipv6) {
+      if (ipv6) {
         // IPv6 multicast addresses start with 0xFF00
-        if (listener.team.address.split(':')[0].toUpperCase() === 'FF00') {
-          listener.server.addMembership(listener.team.address);
+        if (team.address.split(':')[0].toUpperCase() === 'FF00') {
```

Comment:
This isn't a valid way to check, IPv6 addresses might look like FF:00::1, or even FF::1 is a valid multicast address